### PR TITLE
Wip/lapi pkg/apiserver tests

### PR DESCRIPTION
**Pull request by buixor**
- tests all filters of `GET /alerts`
 - tests all filters of `GET /decisions`
 - Improved tests on `DELETE /decisions`
 - fixed missing scneario version/hash in `models.Alerts` (now version appears correctly in cscli)
 - fixed minor bugs in `DELETE /decisions` and some filters of `GET` on alerts/decisions
 - made name of `bouncers` and `machines` unique

### CODE REVIEW

File: pkg/apiserver/alerts_test.go [hunk 1]
```diff
@@ -93,8 +93,290 @@ func TestCreateAlert(t *testing.T) {
...
+	//test until
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("GET", "/v1/alerts?until=1zuzu", nil)
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 500, w.Code)
+	assert.Equal(t, w.Body.String(), `{"message":"while parsing duration: time: unknown unit zuzu in duration 1zuzu"}`)
+
+	//test simulated
```
Comment (erenJag) [inline pkg/apiserver/alerts_test.go:304]: Maybe you need to add 2 decisions on the DB, one with simulated and the other one without.

File: pkg/apiserver/alerts_test.go [hunk 2]
```diff
@@ -93,8 +93,290 @@ func TestCreateAlert(t *testing.T) {
...
+		log.Fatalln(err.Error())
+	}
+
+	alertContentBytes, err := ioutil.ReadFile("./tests/alert_bulk.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	alertContent := string(alertContentBytes)
+
+	w := httptest.NewRecorder()
```
Comment (erenJag) [inline pkg/apiserver/alerts_test.go:366]: Please add context in comment above ass you done for other tests

File: pkg/apiserver/alerts_test.go [hunk 3]
```diff
@@ -93,8 +93,290 @@ func TestCreateAlert(t *testing.T) {
...
+	req, _ := http.NewRequest("POST", "/v1/alerts", strings.NewReader(alertContent))
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+
+	w = httptest.NewRecorder()
```
Comment (erenJag) [inline pkg/apiserver/alerts_test.go:372]: context here also

File: pkg/apiserver/alerts_test.go [hunk 4]
```diff
@@ -117,25 +399,50 @@ func TestListAlert(t *testing.T) {
...
+		log.Fatalln(err.Error())
+	}
+
+	alertContentBytes, err := ioutil.ReadFile("./tests/alert_sample.json")
+	if err != nil {
+		log.Fatal(err)
+	}
+	alertContent := string(alertContentBytes)
+
+	w := httptest.NewRecorder()
```
Comment (erenJag) [inline pkg/apiserver/alerts_test.go:432]: context here plz

File: pkg/apiserver/alerts_test.go [hunk 5]
```diff
@@ -117,25 +399,50 @@ func TestListAlert(t *testing.T) {
...
+	req, _ := http.NewRequest("POST", "/v1/alerts", strings.NewReader(alertContent))
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", "ratata"))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 401, w.Code)
+
+	w = httptest.NewRecorder()
```
Comment (erenJag) [inline pkg/apiserver/alerts_test.go:439]: context here plz

File: pkg/apiserver/decisions_test.go [hunk 6]
```diff
@@ -12,6 +12,176 @@ import (
...
+	// delete by ip
+	w = httptest.NewRecorder()
+	req, _ = http.NewRequest("DELETE", "/v1/decisions?range=91.121.79.0/24", strings.NewReader(""))
+	req.Header.Add("User-Agent", UserAgent)
+	req.Header.Add("Authorization", fmt.Sprintf("Bearer %s", loginResp.Token))
+	router.ServeHTTP(w, req)
+	assert.Equal(t, 200, w.Code)
+	assert.Equal(t, `{"nbDeleted":"2"}`, w.Body.String())
+
+	// delete by scope/value
```
Comment (erenJag) [inline pkg/apiserver/decisions_test.go:52]: context is different than the test

File: pkg/database/alerts.go [hunk 7]
```diff
@@ -320,14 +209,6 @@ func BuildAlertRequestFromFilter(alerts *ent.AlertQuery, filter map[string][]str
 				return nil, fmt.Errorf("Empty time now() - %s", since.String())
 			}
 			alerts = alerts.Where(alert.CreatedAtGTE(since))
-		case "simulated":
```
Comment (erenJag) [inline pkg/database/alerts.go:323]: why the simulated is deleted from the request builder ?

File: pkg/apiserver/alerts_test.go (hunk 1 above)
Comment (buixor) [reply pkg/apiserver/alerts_test.go:342]: added a dedicated test `TestSimulatedAlert` with a specific dataset testing the `simulated` filter

File: pkg/apiserver/decisions_test.go (hunk 6 above)
Comment (buixor) [reply pkg/apiserver/decisions_test.go:52]: fixed commens if it was what you meant ?

**Review by erenJag — APPROVED** [review]

**Pull request merged**